### PR TITLE
Issue #10932: remove Powermock from HeaderTest

### LINKED ISSUE #9987: Resolve Pitest Issues - AbstractHeaderCheck (1)
Child Issue of #7797 ,
https://github.com/checkstyle/checkstyle/blob/master/.ci/pitest.sh#L54

### PR DESCRIPTION
**Pull request by pbludov**
Issue #10932

The access modifier for the method `loadHeader` is `package` now. It is not possible to mock private method anymore.

As a side effect, closes #9987

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java [hunk 1]
```diff
@@ -171,11 +171,12 @@ public void setHeader(String header) {
 
     /**
      * Load header to check against from a Reader into readerLines.
+     * This method has package access modifier because it is used in tests.
      *
      * @param headerReader delivers the header to check against
      * @throws IOException if
      */
-    private void loadHeader(final Reader headerReader) throws IOException {
+    /* package */ void loadHeader(final Reader headerReader) throws IOException {
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java:179]: This is the only possible way. See https://github.com/mockito/mockito/wiki/Mockito-And-Private-Methods

**Review by strkkk — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java:179]: This worries me a lot ... We did a lot of work to remove package and protected visibility just testing.

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java:179]: I have an idea. Maybe it is possible to implement a mock chain to simulate an exception in the `loadHeader` method.

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java:179]: Succeeded with
```java
            final Exception ex = assertThrows(Exception.class,
                    () -> {
                        try (MockedConstruction<LineNumberReader> mocked = mockConstruction(
                                LineNumberReader.class, (mock, context) -> {
                                    when(mock.readLine()).thenThrow(IOException.class);
                                })) {
                            check.setHeader("header");
                        }
                    },
                    "IllegalArgumentException expected");
```
This PR will be closed, the changes will be in the PR #10947

**Pull request closed without merging**